packages/pigeon/lib/java_generator.dart

```diff
@@ -152,11 +153,22 @@ void _writeHostApi(Indent indent, Api api) {
     final String channelName = makeChannelName(api, method);
     indent.write('');
     indent.scoped('{', '}', () {
+      String? taskQueue;
+      if (method.taskQueueType != TaskQueueType.serial) {
+        taskQueue = 'taskQueue';
+        indent.writeln(
+            'BinaryMessenger.TaskQueue taskQueue = binaryMessenger.makeBackgroundTaskQueue();');
+      }
       indent.writeln('BasicMessageChannel<Object> channel =');
       indent.inc();
       indent.inc();
-      indent.writeln(
-          'new BasicMessageChannel<>(binaryMessenger, "$channelName", getCodec());');
+      indent.write(
+          'new BasicMessageChannel<>(binaryMessenger, "$channelName", getCodec()');
+      if (taskQueue != null) {
+        indent.addln(', $taskQueue);');
+      } else {
+        indent.addln(');');
+      }
```

Comment:
Why not `taskQueueParameter = ', taskQueue';` above in place of `taskQueue`, then just inline that into the `writeln` abowe, so that the closing paren isn't split out and in a conditional?

Reply:
In the past I've done that and the extra layer of indirection of defining a variable may have made the code a bit harder to follow than to have a conditional.  I switched it to the conditional here.  I'm still not sure if one way is more clear than the other.